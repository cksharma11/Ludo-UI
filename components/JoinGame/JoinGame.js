import React from 'react';
import Header from '../Header/Header';
import JoinGameStyles from './JoinGame.style';
import labels from '../../config/labels/labels';

const JoinGame = () => {
  return (
    <div className="container">
      <Header title="Ludo" />
      <div className="form-container">
        <form method="POST" action="/joinGame">
          <div className="activity-header">{labels.JOIN_GAME}</div>
          <input type="text" name="playerName" placeholder="name" />
          <input type="number" name="gameId" placeholder="gameId" />
          <input type="submit" value={labels.JOIN_GAME} />
        </form>
      </div>
      <style jsx>{JoinGameStyles}</style>
    </div>
  );
};

export default JoinGame;
