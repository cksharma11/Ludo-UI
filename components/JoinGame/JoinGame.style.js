import css from 'styled-jsx/css';
import colors from '../../config/styles/colors';

const JoinGameStyles = css`
  .form-container {
    display: flex;
    justify-content: center;
    align-items: center;
    height: 660px;
  }

  form {
    display: flex;
    height: 550px;
    width: 350px;
    flex-direction: column;
    justify-content: center;
    align-items: center;
    background: white;
    border: 1px solid ${colors.gray.shade2};
    font-family: 'Ubuntu', sans-serif;
  }

  form > input {
    background: ${colors.gray.shade3};
    width: 200px;
    height: 30px;
    border: 1px solid ${colors.gray.shade1};
    color: black;
    text-align: center;
    line-height: 2;
    text-decoration: none;
    padding: 5px;
    margin: 5px;
    border-radius: 2px;
    font-family: 'Ubuntu', sans-serif;
  }

  form > input:hover {
    background: white;
  }

  .activity-header {
    font-size: 30px;
  }

  .container {
    width: 100vw;
    height: 100vh;
    @import url('https://fonts.googleapis.com/css?family=Ubuntu:300&display=swap');
    padding: 0;
    margin: -8px;
    background: radial-gradient(
      ellipse at center,
      rgb(99, 155, 155) 0%,
      rgb(25, 41, 41) 99%,
      rgb(69, 108, 108) 100%
    );
    font-family: 'Ubuntu', sans-serif;
  }
  button {
    font-family: 'Ubuntu', sans-serif;
    border-radius: 3px;
  }
`;

export default JoinGameStyles;
