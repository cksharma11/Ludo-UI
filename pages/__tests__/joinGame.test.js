import React from 'react';
import { shallow } from 'enzyme';
import JoinGame from '../joinGame';

describe('joinGame', () => {
  it('should render properly', () => {
    const wrapper = shallow(<JoinGame />);
    expect(wrapper.debug()).toMatchSnapshot();
  });
});
