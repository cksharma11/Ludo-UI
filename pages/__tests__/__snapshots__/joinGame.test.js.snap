// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`joinGame should render properly 1`] = `"<JoinGame />"`;
